near/test/sdk.ts

```diff
@@ -0,0 +1,874 @@
+  );
+
+  let randoToken = nearProviders.getTransactionLastResult(
+    await userAccount.functionCall({
+      contractId: "test.test.near",
+      methodName: "deploy_ft",
+      args: {
+        account: userAccount.accountId,
+      },
+      gas: new BN("300000000000000"),
+    })
+  );
+
+  let token_bridge = CONTRACTS.DEVNET.near.token_bridge;
+  let core_bridge = CONTRACTS.DEVNET.near.core;
+
+  console.log("Setting up algorand wallet");
+
+  let algoCore = BigInt(CONTRACTS.DEVNET.algorand.core);
+  let algoToken = BigInt(CONTRACTS.DEVNET.algorand.token_bridge);
+
+  const tbAddr: string = getApplicationAddress(algoToken);
+  const decTbAddr: Uint8Array = decodeAddress(tbAddr).publicKey;
+  const aa: string = uint8ArrayToHex(decTbAddr);
+
+  const algoClient: algosdk.Algodv2 = getAlgoClient();
+  const tempAccts: Account[] = await getTempAccounts();
+  const numAccts: number = tempAccts.length;
+
+  const algoWallet: Account = tempAccts[0];
+
+  console.log("Creating USDC on Near");
+
+  let ts = new TestLib();
+  let seq = Math.floor(new Date().getTime() / 1000);
+  let usdcvaa = ts.hexStringToUint8Array(
+    ts.genAssetMeta(
+      ts.singleGuardianPrivKey,
+      0,
+      1,
+      seq,
+      "4523c3F29447d1f32AEa95BEBD00383c4640F1b4".toLowerCase(),
+      1,
+      8,
+      "USDC",
+      "CircleCoin"
+    )
+  );
+
+  seq = seq + 1;
+
+  let usdcp = _parseVAAAlgorand(usdcvaa);
+
+  //console.log(usdcp);
+
+  console.log("calling createWrappedOnNear to create usdc");
+
+  if (
+    (await getIsTransferCompletedNear(userAccount, token_bridge, usdcvaa)) ==
+    true
+  ) {
+    console.log("getIsTransferCompleted returned incorrect value (true)");
+    process.exit(1);
+  }
+
+  let usdc = await createWrappedOnNear(userAccount, token_bridge, usdcvaa);
+  console.log(usdc);
+
+  if (usdc === "") {
+    console.log("null usdc ... we failed to create it?!");
+    process.exit(1);
+  }
+
+  if (
+    (await getIsTransferCompletedNear(userAccount, token_bridge, usdcvaa)) ==
+    false
+  ) {
+    console.log("getIsTransferCompleted returned incorrect value (false)");
+    process.exit(1);
+  }
+
+  let aname = await getForeignAssetNear(
+    userAccount,
+    token_bridge,
+    usdcp.FromChain as ChainId,
+    usdcp.Contract as string
+  );
+  if (aname !== usdc) {
+    console.log(aname + " !== " + usdc);
+    process.exit(1);
+  } else {
+    console.log(aname + " === " + usdc);
+  }
+
+  console.log("Creating USDC token on algorand");
+  let tx = await createWrappedOnAlgorand(
+    algoClient,
+    algoToken,
+    algoCore,
+    algoWallet.addr,
+    usdcvaa
+  );
+  await signSendAndConfirmAlgorand(algoClient, tx, algoWallet);
+
+  let account_hash = await userAccount.viewFunction(
+    token_bridge,
+    "hash_account",
+    {
+      account: userAccount.accountId,
+    }
+  );
+
+  console.log(account_hash);
+
+  let myAddress = account_hash[1];
+
+  console.log("Airdropping USDC on myself");
+  {
+    let trans = ts.genTransfer(
```

Comment:
How come you didn't call transferFromNear()?

Reply:
Because I first have to give myself some of this USDC in the first place... so I create the vaa synthetically and then can redeemOnNear to give it to me